homeassistant/components/tradfri/light.py

```diff
@@ -106,41 +88,9 @@ def brightness(self):
             if kwargs[ATTR_BRIGHTNESS] == 255:
                 kwargs[ATTR_BRIGHTNESS] = 254
 
-            await self._api(self._group.set_dimmer(kwargs[ATTR_BRIGHTNESS], **keys))
+            await self._api(self._device.set_dimmer(kwargs[ATTR_BRIGHTNESS], **keys))
         else:
-            await self._api(self._group.set_state(1))
-
-    @callback
-    def _async_start_observe(self, exc=None):
-        """Start observation of light."""
-        if exc:
-            _LOGGER.warning("Observation failed for %s", self._name, exc_info=exc)
-
-        try:
-            cmd = self._group.observe(
-                callback=self._observe_update,
-                err_callback=self._async_start_observe,
-                duration=0,
-            )
-            self.hass.async_create_task(self._api(cmd))
-        except PytradfriError as err:
-            _LOGGER.warning("Observation failed, trying again", exc_info=err)
-            self._async_start_observe()
-
-    def _refresh(self, group):
-        """Refresh the light data."""
-        self._group = group
-        self._name = group.name
-
-    @callback
-    def _observe_update(self, tradfri_device):
-        """Receive new state data for this light."""
-        self._refresh(tradfri_device)
-        self.async_schedule_update_ha_state()
-
-    async def async_update(self):
```

Comment:
I can't see that this method is being used anywhere, or am I missing something?

Reply:
It is used if an entities poll property is true

Reply:
Thus since we're using `async_schedule_update_ha_state`, it's correct to remove it?

Reply:
If you use poll, it needs exists otherwise you can remove it. Default Entity object have enabled poll.

Reply:
Tradfri light groups are polling besides the observations. There's some issue with observations for groups.

Reply:
Adding some info about this to the documentation.